Source/santad/SNTDaemonControlController.mm

```diff
@@ -156,6 +156,35 @@ - (void)staticRuleCount:(void (^)(int64_t count))reply {
   reply([SNTConfigurator configurator].staticRules.count);
 }
 
+- (void)retrieveAllRules:(void (^)(NSArray *, NSError *))reply {
+  SNTConfigurator *config = [SNTConfigurator configurator];
+
+  // Do not return any rules if syncBaseURL is set and return an error
+  if (config.syncBaseURL) {
+    reply(@[], [NSError errorWithDomain:@"com.google.santad"
+                                   code:1  // TODO define error code
+                               userInfo:@{NSLocalizedDescriptionKey : @"SyncBaseURL is set"}]);
+    return;
+  }
+
+  SNTRuleTable *rdb = [SNTDatabaseController ruleTable];
+  NSArray *rules = [rdb retrieveAllRules];
+
+  // Convert Rules to an NSDictionary.
+  NSMutableArray *rulesAsDicts = [[NSMutableArray alloc] init];
+
+  for (SNTRule *rule in rules) {
+    // Omit transitive and remove rules as they're not relevan.
+    if (rule.state == SNTRuleStateAllowTransitive || rule.state == SNTRuleStateRemove) {
+      continue;
+    }
+
+    [rulesAsDicts addObject:[rule dictionaryRepresentation]];
+  }
+
+  reply(rulesAsDicts, nil);
```

Comment:
It might be interesting to consider switching things up here a bit - instead of creating the dictionary in the daemon, reply with the array of rules and do the construction within `santactl`. `SNTRule` already conforms to `NSSecureCoding` so it can be sent over XPC.

This would have the benefit of offloading work out of the daemon and also make this retrieval more versatile for future operations.

Reply:
I agree. I'll make the change.

Reply:
Change made. We're now converting the rules to json in `santactl`.